src/cmd/run.go

```diff
@@ -477,10 +479,10 @@ func constructExecArgs(container string,
 
 	execArgs = append(execArgs, []string{
 		container,
-		"capsh", "--caps=", "--", "-c", "exec \"$@\"", "/bin/sh",
+		"capsh", "--caps=", "--", "-c",
 	}...)
 
-	execArgs = append(execArgs, command...)
+	execArgs = append(execArgs, fmt.Sprintf("exec %s", strings.Join(command, " ")))
```

Comment:
Ditto.  Even more importantly, how is this objectively better than the existing code?  It also seems wrong.

If some of the elements in `command` themselves contain spaces, then `strings.Join(command, " ")` will split them into separate arguments for the `exec`.  That would be the equivalent of `$*` or `$@` without the quotes.  Instead they need to be passed as is, without further splitting.  Hence the use of ["$@" with the quotes](https://stackoverflow.com/questions/21071943/difference-between-and-in-bash-script).

Let's not deal with the escaping of whitespaces and quoting, and leave it to the shell.

Reply:
yeah I think my change was incorrect here